src/core/ViewModels/ApplicationViewModel.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef APPLICATIONVIEWMODEL_HPP
+#define APPLICATIONVIEWMODEL_HPP
+
+#include <QObject>
+#include <memory>
+
+class EmulatorWalletDataSource;
+class WalletDataSource;
+
+class ApplicationViewModel : public QObject
+{
+    Q_OBJECT
+public:
+    explicit ApplicationViewModel(QObject *parent = nullptr);
+    ~ApplicationViewModel();
+
+    WalletDataSource *dataSource();
```

Comment:
1. If data source is stored as `shared_ptr` makes sense to give `std::shared_ptr` or `std::weak_ptr`, but not raw pointer
2. Getters lack `const`